tests/unit_tests/analysis/test_es_update.py

```diff
@@ -351,3 +352,42 @@ def test_update_multiple_param(copy_case):
     # https://en.wikipedia.org/wiki/Variance#For_vector-valued_random_variables
     for prior_name, prior_data in prior.items():
         assert np.trace(np.cov(posterior[prior_name])) < np.trace(np.cov(prior_data))
+
+
+@pytest.mark.integration_test
+def test_gen_data_obs_data_mismatch(copy_case):
```

Comment:
Might be possible to test this on the snake oil case with storage by modifying the observations and then running an update? Then the runtime will be quite a lot faster.